### PR TITLE
all: add dark mode (fixes #3564)

### LINKED ISSUE #3564: dark theme for myPlanet

### PR DESCRIPTION
**Pull request by ewhang5**
draft pull request for making progress towards dark theme

progress:
- functionality for dark mode button in settings
- some additional files created for night mode
- started converting colors for pages

to do:
- app does not check dark mode button selection when restarting (you have to reselect dark mode even if it is selected for the app to be updated after a refresh)
- convert colors for additional pages (check my screenshots sent to Gideon in the discord to see what still needs to be converted), fine tune colors for dark mode
- refactor additional pages in layout-night (these additional pages are often from having to apply different tints for drawable items, we could instead create more drawable items and have them in a drawable-night directory)

### CODE REVIEW

**Comment by Okuro3499** [comment]
@ewhang5 you could also be resolving the conflicts a you work on this so it does not overwhelm you in the future

File: app/src/main/res/layout/row_chat_history.xml [hunk 1]
```diff
@@ -5,6 +5,9 @@
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
     android:layout_gravity="top"
```
Comment (Okuro3499) [inline app/src/main/res/layout/row_chat_history.xml:7]: we have 2 `android:id="@+id/chatCardView"` causing the crash

**Review by Okuro3499 — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/row_chat_history.xml (hunk 1 above)
Comment (ewhang5) [reply app/src/main/res/layout/row_chat_history.xml:7]: I saw this as well. Getting rid of it now

**Review by Okuro3499 — CHANGES_REQUESTED** [review]
settings text is not visible in dark theme
Screenshot_20240625_215615

**Review by Okuro3499 — APPROVED** [review]

**Comment by ewhang5** [comment]
@strawberrybread @rlam20 I was wondering if you guys could review this pr. Some of the features are not finished (for example the settings dropdown in dark mode is still white, etc.). For testing, it would be great if you could just make sure that it would be safe to merge the current progress and that there are no blaring issues.

File: app/src/main/java/org/ole/planet/myplanet/MainApplication.kt [hunk 2]
```diff
@@ -70,6 +72,9 @@ class MainApplication : Application(), Application.ActivityLifecycleCallbacks {
 
         context = this
         preferences = getSharedPreferences(PREFS_NAME, MODE_PRIVATE)
+        nightMode()
+        // UNCOMMENT BELOW TO FORCE DARK MODE FOR DARK MODE DEVELOPMENT
+        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
```
Comment (ewhang5) [inline app/src/main/java/org/ole/planet/myplanet/MainApplication.kt:77]: If you wanted to test the light mode functionality, you will have to comment this line out. (I would actually recommend doing this as you can still switch to dark mode functionality via settings if this line is commented out).

**Comment by Okuro3499** [comment]
@dogi build error is no longer occurring

**Review by strawberrybread — APPROVED** [review]

**Review by rlam20 — COMMENTED** [review]
ignore this comment

**Review by rlam20 — APPROVED** [review]

**Review by dogi — APPROVED** [review]
@ewhang5 good job

**Pull request merged**